### PR TITLE
Async merge

### PR DESCRIPTION
**Pull request by ohadbitt**
Async merge so that it doesnt need to keep connection

### CODE REVIEW

File: connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoReader.scala [hunk 1]
```diff
@@ -244,6 +244,7 @@ private[kusto] class KustoReader(client: Client, request: KustoReadRequest, stor
     val commandResult: KustoResultSetTable = client.execute(request.kustoCoordinates.database,
       exportCommand,
       request.clientRequestProperties.orNull).getPrimaryResults
-    KDSU.verifyAsyncCommandCompletion(client, request.kustoCoordinates.database, commandResult, directory,timeOut = request.timeout)
+    KDSU.verifyAsyncCommandCompletion(client, request.kustoCoordinates.database, commandResult, timeOut = request
+      .timeout, doingWhat = s"export data to  blob directory: ('$directory') preparing it for reading.")
```
Comment (yogilad) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoReader.scala:248]: nit: 
Extra space between "to" and "blob"
"in preparation for reading"?

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala [hunk 2]
```diff
@@ -63,11 +63,23 @@ private[kusto] object CslCommandsGenerator {
        |  distinct ExtentId"""
   }
 
+  def generateTableMoveExtentsAsyncCommand(sourceTableName: String, destinationTableName: String, batchSize:Int): String
+  = {
+    s""".move async extents to table $destinationTableName <|
+       .show table $sourceTableName extents with(extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":$batchSize}');
+        $$command_results
```
Comment (yogilad) [inline connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala:70]: What's command_results (not an argument like the rest)?

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala [hunk 3]
```diff
@@ -63,11 +63,23 @@ private[kusto] object CslCommandsGenerator {
...
+       |  distinct ExtentId"""
```
Comment (yogilad) [inline connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala:71]: You'll get a list of only unique extent ids.
* Don't you need the operationIds?
* Why do you need to call distinct? where are the duplications?

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala (hunk 2 above)
Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala:70]: command results is a KQL macro in command-then-query flow - to get the command results as a Table and operate on it

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala (hunk 3 above)
Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala]: this is one operation 
i dont need distinct anymore - this command changed a while ago - will fix

**Review by yogilad — APPROVED** [review]
Check this well since I'm not the authority on the Java / Spark SDK.
You can also want to get another look from Asaf or Yihezkel

**Pull request merged**